backend/inspirehep/disambiguation/utils.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+
+import structlog
+from flask import url_for
+from inspire_dojson.utils import get_record_ref
+
+from inspirehep.records.api.authors import AuthorsRecord
+from inspirehep.records.api.literature import LiteratureRecord
+
+LOGGER = structlog.getLogger()
+
+
+def link_signature_to_author(signature_data, author_control_number):
+    """Adds record/$ref of the given author to the given signature.
+
+    Args:
+        author_control_number (int): The control number of the author to which we want to link.
+        signature_data (list): List containing 2 elements: the publication_id and the signature uuid.
+
+    Returns:
+        dict: The signature data from the publication with the linked author.
+    """
+    record = LiteratureRecord.get_record_by_pid_value(signature_data["publication_id"])
+    signature = next(
+        (
+            author
+            for author in record.get("authors")
```

Comment:
As I said in an earlier comment, you can replace everywhere `.get(foo)` by `[foo]` as you know you're getting valid input.